homeassistant/core.py

```diff
@@ -472,7 +472,8 @@ def set(self, entity_id, new_state, attributes=None):
             old_state = self._states.get(entity_id)
 
             is_existing = old_state is not None
-            same_state = is_existing and old_state.state == new_state
+            same_state = (is_existing and old_state.state == new_state and
+                          not push_state)
```

Comment:
I think it would be _slightly_ cleaner if we move the check if we should force update to the if-statement below.